### PR TITLE
Add gauge chart

### PR DESCRIPTION
**Pull request by JAV04**

### CODE REVIEW

File: WebServer/gauge.html [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+    <script src="https://ajax.googleapis.com/ajax/libs/jquery/2.0.0/jquery.min.js"></script>
+	<script src="https://code.highcharts.com/highcharts.js"></script>
+	<script src="https://code.highcharts.com/highcharts-more.js"></script>
+	<script src="https://code.highcharts.com/modules/solid-gauge.js"></script>
+	<script src="db.js"></script>
+    <script src="theme.js"></script>
+
+  </head>
+  <body>
+        <script type="text/javascript">
```
Comment (BenMatase) [inline WebServer/gauge.html]: unecessary extra script tag

File: WebServer/gauge.html [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+        </script>
+            <script type="text/javascript">
```
Comment (BenMatase) [inline WebServer/gauge.html]: Then it looks like it is indented extra

File: WebServer/gauge.html [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+
+            //get info for each sensor selected
+            var condition = "WHERE sensorId in (" + result.id + ") ORDER BY sensorId ASC;"
+            var sensorInfo = readDB("sensor",condition,"*");
+
+            //get project info for sensors selected (take first because all sensors will be associated with same project)
+            var condition = "WHERE projectId = " + sensorInfo[0].projectId + " LIMIT 1;"
+            var projectInfo = readDB("project",condition,"*")[0];
+
+			var gaugeOptions = {
```
Comment (BenMatase) [inline WebServer/gauge.html]: This is indented a lot more than surrounding code

File: WebServer/gauge.html [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+                            valueSuffix: sensorInfo[i].units
+                            }, 
+			        }]
+
+		    	}));
+                gaugeCharts.push(chart);
+            }
+            
+
+setInterval(function () {
```
Comment (BenMatase) [inline WebServer/gauge.html]: Indent this

File: WebServer/gauge.html [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+                var initVal = readDB("data",condition,"value")[0];
+
+                //create div element
+                var iDiv = document.createElement('div');
+                iDiv.id = sensorInfo[i].name;
+                iDiv.style = "width: 300px; height: 200px; float: left";
+                document.getElementsByTagName('body')[0].appendChild(iDiv);
+
+                //create chart for sensor
+			    var chart = Highcharts.chart(sensorInfo[i].name, Highcharts.merge(gaugeOptions, {
```
Comment (BenMatase) [inline WebServer/gauge.html]: unindent this

File: WebServer/visualize.html [hunk 6]
```diff
@@ -68,9 +70,9 @@
                 },
                 dateTimeLabelFormats: {
                     millisecond: '%H:%M:%S.%L',
-                    second: '%H:%M:%S',
-                    minute: '%H:%M',
-                    hour: '%H:%M',
+                    second: '%l:%M:%S %p',
```
Comment (BenMatase) [inline WebServer/visualize.html]: What does this do?

File: WebServer/visualize.html [hunk 7]
```diff
@@ -83,24 +85,21 @@
...
             for(var i=0; i< sensorIds.length; i++) {
                 var condition = "WHERE sensorId = " + sensorIds[i] + " ORDER BY dateTime ASC;";
-                var data = readDB("datahourly",condition,"*");
+                var data = readDB("datahourly",condition,"sensorId, averageValue, sampleRate, dateTime");
 
                 var nodes = [] //holds all datapoints for a given series
                 for(var j=0; j<data.length; j++) { 
 
                     var time = data[j].dateTime.split(/[- :]/);; //formatting date for highcharts to understand
-                    var nodeDate = new Date(Date.UTC(time[0], time[1]-1, time[2], time[3], time[4]));
```
Comment (BenMatase) [inline WebServer/visualize.html]: I'm assuming this has to do with the changes on line 73

File: WebServer/visualize.html (hunk 6 above)
Comment (JAV04) [reply WebServer/visualize.html]: was trying to format with 1-12 am/pm vs 1:00-24:00 but that did not work

Comment (BenMatase) [reply WebServer/visualize.html]: Ahhh.  Gotcha.  So this code just settles for the military time format?

**Pull request merged**